dipy/reconst/dki_micro.py

```diff
@@ -0,0 +1,489 @@
+#!/usr/bin/python
+""" Classes and functions for fitting the diffusion kurtosis model """
+from __future__ import division, print_function, absolute_import
+
+import numpy as np
+import dipy.reconst.dki as dki
+from dipy.reconst.dti import (TensorFit, apparent_diffusion_coef,
+                              _positive_evals, eig_from_lo_tri,
+                              _decompose_tensor_nan, trace,
+                              radial_diffusivity, axial_diffusivity,
+                              MIN_POSITIVE_SIGNAL)
+
+from dipy.reconst.dki import (split_dki_param, apparent_kurtosis_coef,
+                              kurtosis_maximum, common_fit_methods,
+                              DiffusionKurtosisModel, DiffusionKurtosisFit)
+from dipy.reconst.utils import dki_design_matrix as design_matrix
+from dipy.reconst.dti import design_matrix as dti_design_matrix
+from dipy.reconst.base import ReconstModel
+from dipy.core.ndindex import ndindex
+from dipy.reconst.vec_val_sum import vec_val_vect
+
+
+def axonal_water_fraction(dki_params, sphere='symmetric362', gtol=1e-5,
+                          mask='None'):
+    """ Computes the axonal water fraction from DKI [1]_.
+
+    Parameters
+    ----------
+    dki_params : ndarray (x, y, z, 27) or (n, 27)
+        All parameters estimated from the diffusion kurtosis model.
+        Parameters are ordered as follow:
+            1) Three diffusion tensor's eingenvalues
+            2) Three lines of the eigenvector matrix each containing the first,
+               second and third coordinates of the eigenvector
+            3) Fifteen elements of the kurtosis tensor
+    sphere : Sphere class instance, optional
+        The sphere providing sample directions for the initial search of the
+        maximal value of kurtosis.
+    gtol : float, optional
+        This input is to refine kurtosis maxima under the precision of the
+        directions sampled on the sphere class instance. The gradient of the
+        convergence procedure must be less than gtol before successful
+        termination. If gtol is None, fiber direction is directly taken from
+        the initial sampled directions of the given sphere object
+    mask : ndarray
+        A boolean array used to mark the coordinates in the data that should be
+        analyzed that has the shape dki_params.shape[:-1]
+
+    Returns
+    --------
+    AWF : ndarray (x, y, z) or (n)
```

Comment:
This was changed everywhere except right here in the docstring. Please change docstring accordingly.